code/go/0chain.net/smartcontract/storagesc/allocation.go

```diff
@@ -426,7 +413,7 @@ func (sc *StorageSmartContract) selectBlobbers(
 	// size of allocation for a blobber
 	var bSize = (sa.Size + int64(size-1)) / int64(size)
 	var list = sa.filterBlobbers(allBlobbersList.Nodes.copy(), creationDate,
-		bSize, filterHealthyBlobbers(creationDate),
```

Comment:
@Sriep Maybe I misunderstood. My idea is to pass the list of blobbers (urls of blobbers) in the allocation endpoint, then this filter is picking up blobbers by URL and a few other filters like healthy, size etc. In any case, I will need to iterate through blobbers to get Blobber object?

Reply:
A list of 100,000+ blobbers is too much to keep in the MPT. Takes over 50ms to get and update. So that is being removed soon. So pretend the allBlobbers list is not there, but leave the existing code, it will be removed shortly.

Instead, the list of blobbers checked here has to be loaded individually. Updating 40 items from the MPT is bad, but probably only about 10-20ms so we can handle it. 

I can's see any reason the same `filterBlobbers` can't be used to check the user-selected list of blobbers the same way it filtered the all blobbers list.

We want to avoid doing anything that involves manipulating hundred of thousands of blobbers.

Reply:
Actually, if you could remove the all blobbers list entirely from new allocation request, that would help.

Reply:
I just modified https://github.com/0chain/0chain/pull/691 to supply `/getBlobbers` with its information. I believe that the [all blobbers list](https://github.com/0chain/0chain/blob/staging/code/go/0chain.net/smartcontract/storagesc/blobber.go#L17) is only used in new allocation requests and [/getBlobbes](https://github.com/0chain/0chain/blob/blobber-events/code/go/0chain.net/smartcontract/storagesc/handler.go#L53). The other references are just maintaining the list. 

Any endpoint that maintains the all blobbers list has to take >50ms with 100,000+ blobbers so removing it will be a solid step towards making our smartcontracts scalable.